rfcs/1-jhipster-rfc-k8s-operator.md

```diff
@@ -0,0 +1,161 @@
+// This is a RFC template based on the Rust RFC process but simplified: https://github.com/rust-lang/rfcs/
+
+- Feature Name: JHipster Kubernetes Operator
+- Start Date: 8/8/19
+- Issue: [jhipster/generator-jhipster#10053](https://github.com/jhipster/generator-jhipster/10053)
+
+# Summary
+[summary]: #summary
+
+The JHipster Operators uses Kubernetes CRDs (Custom Resource Definitions) to encapsulate operational knowledge about JHipster Applications. 
+
+
+# Motivation
+[motivation]: #motivation
+
+Integrating with Kubernetes requires more than just the Kubernetes Manifest to get our containers running inside the platform. The JHipster Kubernetes Operator will enable JHipster MicroService Applications (projects) to be managed and understood by the platform in a native way.
+
+**Note:** It is important to understand that the current JHipster architecture is not going to be changed. A Kubernetes Operator is something that will run along our JHipster Applications. 
+
+The JHipster Kubernetes Operator adds on top of existing JHipster Applications running in K8s all the operational knowledge required to manage and maintain these applications over time. 
+
+The whole idea of having an operator makes sense only after having an application running in Kubernetes for some time. In other words, if you are looking for just deploying your application you should use the `jhipster kubernetes` or `jhipster kubernetes-helm` generators. 
+
+Once your application is running there will be questions that start to pop up such as:
+
+- What happens if I want to deploy a second (or more) JHipster MicroServices Application? Is it going to work? 
+  - If I have two (or more) applications , can I share the JHipster Registry between the two? 
+- How can I make sure that all the MicroServices that are composing my application are up? How can I be notified if something goes down? 
+- How can I upgrade/version an entire application (Set of microservices)?
+
+With the JHipster Operator we try to provide a place (component) where we can iteratively answer all these questions. 
+
+![JHipster K8s Operator](https://github.com/salaboy/jhipster-operator/blob/master/imgs/jhipster-operator-simplified.png "JHipster K8s Operator")
+
+Notice that all the JHipster applications are running as we know them today inside a K8s cluster, and the JHipster K8s Operator sits on top and with these existing applications. 
+
+Main objectives: 
+- Understand JHipster Applications (root resource)
+  - And Modules (Gateway, MicroServices, Registry..) (secondary/managed resources)
+  - It makes K8s aware of these concepts
+- Manage one or more JHipster Applications  
+- Understand a JHipster Application topology
+  - Understand how things are wired together
+  - Understand and validate the Applications State as a whole
+- Provide advanced lifecycle management
+  - Shared infrastructure wiring
```

Comment:
In this section I would add `Manage centralized configuration`, I would love to have an higher level abstraction to manage configuration in kubernetes. Currently we have the ConfigMap but it's not great for many reasons (for example, it does not trigger an update of the apps that depend on it).

Reply:
Totally.. that is part of lifecycle management and understanding how things are wired together. 
I will add it there.

Reply:
done